src/__test__/api.test.ts

```diff
@@ -89,7 +89,7 @@ describe('API', () => {
       await expect(api['request'](method, url)).resolves.toBeUndefined()
 
       expect(global.fetch).toHaveBeenCalled()
-      expect(consoleErrorSpy).toHaveBeenCalled()
```

Comment:
in addition to unit test, would be good to simulate the network error and 400 response error in functional tests too